src/create/mod.rs

```diff
@@ -0,0 +1,27 @@
+use crate::util::Fallible;
+use std::fs::OpenOptions;
+use std::io::Write;
+
+pub fn create_new_project(main_file: &str) -> Fallible {
+    let main_file = format!("{}.md", main_file);
+
+    let mut document = OpenOptions::new()
+        .write(true)
+        .create_new(true)
+        .open(&main_file)?;
+
+    let mut config = OpenOptions::new()
+        .write(true)
+        .create_new(true)
+        .open("Yarner.toml")?;
+
+    document.write_all(DOCUMENT.as_bytes())?;
```

Comment:
Am I right that the actual files are created here, not in the above two statements already? If so, fine...

Reply:
No, the files are create above which is actually the TOCTOU thing: Between checking if a file exists and then creating and writing to it, there is a race. Hence, this first creates both files, but only if neither does already exist, and only then writes to them.

Reply:
(One other thing one could do is to unlink the document file, if the config file could not be created. This would avoid leaving garbage around in addition to avoiding to overwrite existing content.)

Reply:
But if I understand correctly, it already creates the document, although the config exists? That was the reason for my question.

Reply:
> But if I understand correctly, it already creates the document, although the config exists?

Yes, it does create an empty file even if the configuration already exists. If you just do not the like order of things, I can certainly switch it. Or I could make sure to remove the first file if we fail to create the second one, I whichever order. But creating the first before checking for existence of the second is necessary in principle, because the first could be created while checking the second otherwise, i.e. TOCTOU.

Reply:
How about now?